src/common/transformations/src/transformations/common_optimizations/split_squeeze_concat_fusion.cpp

```diff
@@ -120,3 +107,40 @@ ov::pass::SplitSqueezeConcatFusion::SplitSqueezeConcatFusion() {
     auto m = std::make_shared<ov::pass::pattern::Matcher>(concat_pattern, matcher_name);
     register_matcher(m, callback);
 }
+
+bool is_axis_squeezed_by_node(const std::shared_ptr<ov::Node>& squeeze_node, int64_t axis) {
+    const auto& input_shape = squeeze_node->get_input_partial_shape(0);
+    const auto& output_shape = squeeze_node->get_output_partial_shape(0);
+    if (input_shape.rank().is_dynamic() || output_shape.rank().is_dynamic())
+        return false;
+
+    auto input_rank = input_shape.rank().get_length();
+    auto output_rank = output_shape.rank().get_length();
+    // check if output_rank == input_rank - 1
+    // to make sure the node actually squeezes a dimension
+    if (input_rank != output_rank + 1)
+        return false;
+
+    // check if squeezed dimension equals to 1
+    if (input_shape[axis].is_dynamic() || input_shape[axis] != 1)
+        return false;
+
+    // check if the dimensions surrounding squeezed axis matches:
+    // input shape = [..., a, 1, b, ...]
+    // output shape = [..., x, y, ...]
+    // function returns false if a != x or b != y
+    if (axis > 0) {
```

Comment:
I believe we need to check all dims, not only x,y

Maybe this approach is better:
1. delete dim at axis from input_shape
2. compare all dims in input_shape and output_shape, they have to be equal

Reply:
done